metadata-models/src/main/pegasus/com/linkedin/ml/aspect/ModelAspect.pdl

```diff
@@ -0,0 +1,36 @@
+namespace com.linkedin.ml.aspect
+
+import com.linkedin.common.InstitutionalMemory
+import com.linkedin.common.Ownership
+import com.linkedin.common.Status
+import com.linkedin.ml.metadata.CaveatsAndRecommendations
+import com.linkedin.ml.metadata.EthicalConsiderations
+import com.linkedin.ml.metadata.EvaluationData
+import com.linkedin.ml.metadata.IntendedUse
+import com.linkedin.ml.metadata.Metrics
+import com.linkedin.ml.metadata.ModelFactors
+import com.linkedin.ml.metadata.ModelProperties
+import com.linkedin.ml.metadata.QuantitativeAnalyses
+import com.linkedin.ml.metadata.TrainingData
+import com.linkedin.common.Cost
+import com.linkedin.common.Deprecation
+
+/**
+ * A union of all supported metadata aspects for a ML Model
+ */
+typeref ModelAspect = union[
```

Comment:
General comment: Most of the fields in the aspects are free form strings without clear guidelines on the format of the strings or what are valid/invalid values. This can result in a lot of complex business logic in the api layer. Also, producers of this metadata need to understand exactly what goes in each field and in what format. We need to strike a balance between making the models generic but still opinionated enough to be useful to the rest of the community.

A few recommendations are to use enums to define valid value types or break down each field at an even more granular level to strongly type them as much as possible. If any field really needs to be free form, we need documentation and guidelines on the intended usage, valid values and invalid values.